### DISCUSSION TITLE
Disable local authentication

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14700] [Help Wanted!]
Is it possible to disable local authn, after setting up remote authn?  I know I can delete all the local user accounts, but I'd prefer keeping a local admin account and disabling it in the settings file.  That way I can SSH into the server and re-enable it, in case something ever goes wrong with the auth provider

I'd also like to change the unauthorized redirect to `/oauth/login/oidc/` instead of `/login/`.

**Comment by CTV-2023** [comment]
Some informations here, the request was hiding the local user sign-in fields, quite related to your need
https://github.com/netbox-community/netbox/issues/11672

For the last part, I asked something similar : see https://github.com/netbox-community/netbox/issues/14540, not yet accepted

**Comment by markkuleinio** [comment]
Note that you can also set the local admin password to something that is very hard and not even saved anywhere, and in the case of auth provider problems you can just reset the admin password (using `python manage.py changepassword`). Not exactly same as disabling the admin account but may be enough for some cases.

**Reply by llamafilm** [reply]
I would strongly prefer hiding the local login page completely, to avoid user confusion about which one to use.

**Reply by abhi1693** [reply]
If you are alright with using django shell, it's considered best practice to set the password to non usuable one. Here is a code snippet for it. Ref: https://docs.djangoproject.com/en/5.0/ref/contrib/auth/#django.contrib.auth.models.User.set_unusable_password

```python
from django.contrib.auth.models import User

user = User.objects.get(id=123)
user.set_unusable_password()
user.save()
```

**Reply by candlerb** [reply]
Can you not just remove the admin user (or clear their "is_admin" flag?)

**Reply by llamafilm** [reply]
Removing the admin user sounds fine to me. If the SSO provider ever goes down, you could SSH into the box and run `manage.py createsuperuser`.
